_posts/2022-03-16-datafusion-contrib.md

```diff
@@ -0,0 +1,177 @@
+
+## DataFusion-Tokomak
+
+This [crate](https://github.com/datafusion-contrib/datafusion-tokomak) provides an e-graph based DataFusion optimization framework based on the Rust [egg](https://egraphs-good.github.io) library.  An e-graph is a data structure that powers the equality saturation optimization technique.
+
+As context, the optimizer framework within DataFusion is currently under review with the objective of implementing a more strategic long term solution that is more efficient and simpler to develop.
+
+Some of the benefits of using `egg` within DataFusion are:
+
+- Implements optimized algorithms that are hard to match with manually written optimization passes
+- Makes it easy and less verbose to add optimization rules
+- Plugin framework to add more complex optimizations
+- Egg does not depend on rule order and can lead to a higher level of optimization by being able to apply multiple rules at the same time until it converges
+- Allows for cost-based optimizations
+
+This is an exciting new area for DataFusion with lots of opportunity for community involvement!
+
+## DataFusion-Tui
+
+[DataFusion-tui](https://github.com/datafusion-contrib/datafusion-tui) aka `dft` provides a feature rich terminal application for using DataFusion.  It has drawn inspiration and several features from `datafusion-cli`.  In contrast to `datafusion-cli` the objective of this tool is to provide a light SQL IDE experience for querying data with DataFusion.  This includes features such as the following which are currently implemented:
+
+- Tab Management to provide clean and structured organization of DataFusion queries, results, `ExecutionContext` information, and logs
+  - SQL Editor
+    - Text editor for writing SQL queries
+  - Query History
+    - History of executed queries, their execution time, and the number of returned rows
+  - `ExecutionContext` information
+    - Expose information on which physical optimizers are used and which `ExecutionConfig` settings are set
+  - Logs
+    - Logs from `dft`, DataFusion, and any dependent libraries
+- Support for custom `ObjectStore`s
+  - S3
+- Preload DDL from `~/.datafusionrc` to enable having local "database" available at startup
+
+### Upcoming Enhancements
+
+- SQL Editor
+  - Command to write query results to file
+  - Multiple SQL editor tabs
+- Expose more information from `ExecutionContext`
+- A new help tab that provides information on functions
+- Query custom `TableProvider`s such as [DeltaTable](https://github.com/delta-io/delta-rs) or [BigTable](https://github.com/datafusion-contrib/datafusion-bigtable)
+
+## DataFusion-Java
+
+This [project](https://github.com/datafusion-contrib/datafusion-java) created an initial set of Java bindings to DataFusion.  The project is currently in maintenance mode and is looking for maintainers to drive future development.
+
+# How to Get Involved
+
+If you are interested in contributing to DataFusion, and learning about state of
```

Comment:
I feel like we should plug / suggest how to add new community contributions here too 🤔